src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -2956,20 +2956,15 @@ private void AddDeclaredNontypeMembers(DeclaredMembersAndInitializersBuilder bui
                         break;
 
                     case SyntaxKind.RecordDeclaration:
+                    case SyntaxKind.RecordStructDeclaration:
                         var recordDecl = (RecordDeclarationSyntax)syntax;
                         noteRecordParameters(recordDecl, recordDecl.ParameterList, builder, diagnostics);
                         AddNonTypeMembers(builder, recordDecl.Members, diagnostics);
-                        break;
-
-                    case SyntaxKind.RecordStructDeclaration:
-                        var recordStructDecl = (RecordStructDeclarationSyntax)syntax;
-                        var parameterList = recordStructDecl.ParameterList;
-                        noteRecordParameters(recordStructDecl, parameterList, builder, diagnostics);
-                        AddNonTypeMembers(builder, recordStructDecl.Members, diagnostics);
 
                         // We will allow declaring parameterless constructors
                         // Tracking issue https://github.com/dotnet/roslyn/issues/52240
-                        if (parameterList?.ParameterCount == 0)
+                        var parameterList = recordDecl.ParameterList;
```

Comment:
>var parameterList = recordDecl.ParameterList; [](http://example.com/codeflow?start=24&length=45)

If we are introducing this local unconditionally we might as well use it for the code above.